.github/workflows/cross_platform_ci.yml

```diff
@@ -0,0 +1,101 @@
+name: Build Moveit Core on all platforms
+
+on:
+  workflow_dispatch:
+  pull_request:
+  push:
+    branches:
+      - master
+      - "[kmn]*-devel"
+
+jobs:
+  run_moveit_core_compilation:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+    steps:
+    - uses: actions/checkout@v2
+    - name: Build Moveit Core on Unix
+      if: runner.os == 'Linux' || runner.os == 'macOS'
+      shell: bash -l -eo pipefail {0}
+      run: |
+        export CTEST_OUTPUT_ON_FAILURE=1
+        cd moveit_core
+        ls -al
+
+    - name: Set up Dependencies
+      uses: mamba-org/provision-with-micromamba@main
+      with:
+        environment-file: .github/ci_env.yml
+
+    - name: Build Moveit Core on Unix
+      if: runner.os == 'Linux' || runner.os == 'macOS'
+      shell: bash -l -eo pipefail {0}
+      run: |
+        export CTEST_OUTPUT_ON_FAILURE=1
+        cd moveit_core
+
+        mkdir build
+        cd build
+
+        cmake .. -DCMAKE_PREFIX_PATH=$CONDA_PREFIX \
+                 -DCMAKE_INSTALL_PREFIX=$CONDA_PREFIX \
+                 -DCMAKE_BUILD_TYPE=Release \
+                 -DCATKIN_SKIP_TESTING=OFF \
+                 -G "Ninja"
+
+        ninja
+        ninja run_tests
```

Comment:
I'm sure we don't need to include a Linux build in this workflow at all. We have full Linux CI setup for everything we need.

Reply:
I would like to see the Linux build based on RoboStack included in the new workflow. It would be much nicer to catch possible issues early on in the new workflow, rather than in our periodic rebuilds in the RoboStack repo. For example, our workflow was able to detect https://github.com/ros-planning/moveit/pull/2473.
Both @wolfv and I would be happy to address any potential issues that arise from the new workflow on Linux build directly before merging in the relevant PR - it would be much less work for us than post-hoc fixes.
However, if you feel strongly about it, at least I'd be happy to remove the Linux build if that increases the chances of getting the PR merged so we can at least test on Win+macOS.